keras/layers/recurrent.py

```diff
@@ -1267,6 +1267,14 @@ def __init__(self, units,
 
     def build(self, input_shape):
         input_dim = input_shape[-1]
+
+        if type(self.recurrent_initializer).__name__ == 'Identity':
```

Comment:
Use `isinstance` for robustness (since the initializer module is already imported)

Reply:
You should also restrict this behavior to the fused kernel case, for simplicity

Reply:
By checking on the input shape ratio given to the build method?